developer_docs/how_to_create_a_plugin.asciidoc

```diff
@@ -1678,6 +1678,42 @@ The settings are strongly typed and you have to define it.
 The basic types supported by Foreman are: `:boolean`, `:integer`, `:float`, `:string`, `:text`, `:hash`, `:array`.
 The `:text` type supports markdown and usage of such setting should expect markdown syntax when using it.
 
+==== Validate setting value
+
+To validate setting value, you can use API, that tries to mimic the API of ActiveRecord.
+We can use most of the perks offered by ActiveRecord, only defining on setting name instead of attribute.
+We are adding just some shothands like direct regexp validations.
+The attribute is always `value`, you can't validate anything else as it is the only user input.
+Custom validation methods allow you to run validations against other attributes, but please dont! :)
+
+You have two ways of define the validations:
+* inline with setting definition
+* using `validates` of `validates_with` methods
```

Comment:
Can you make `validates` and `validates_with` methods links? I think they're a Rails concept, but as a new developer I may not know that. Totally not speaking for myself right now ;)

Reply:
I'd rather not as they are mimicing the Rails methods with naming, but instead of `validates :attribute` we have `validates :setting_name` I'm not sure if links would not be missleading, though I can mention the specifics :thinking:

Reply:
Explaining that difference is actually a good thing IMHO.